Use port above 1024 instead of 80 in scdf-server deployment

**Problem description:**
`/src/kubernetes/server/server-deployment.yaml` does not run

```
...
2022-11-03 05:28:38.153  INFO 1 --- [           main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.53]
2022-11-03 05:28:38.655  INFO 1 --- [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2022-11-03 05:28:40.648  INFO 1 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
2022-11-03 05:28:41.456  INFO 1 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Start completed.
2022-11-03 05:28:41.955  INFO 1 --- [           main] o.f.c.internal.license.VersionPrinter    : Flyway Community Edition 7.7.3 by Redgate
2022-11-03 05:28:42.154  INFO 1 --- [           main] o.f.c.i.database.base.DatabaseType       : Database: jdbc:mysql://10.233.14.223:3306/mysql (MySQL 5.7)
2022-11-03 05:28:42.271  INFO 1 --- [           main] o.f.core.internal.command.DbValidate     : Successfully validated 4 migrations (execution time 00:00.012s)
2022-11-03 05:28:42.351  INFO 1 --- [           main] o.f.core.internal.command.DbMigrate      : Current version of schema `mysql`: 4
2022-11-03 05:28:42.352  INFO 1 --- [           main] o.f.core.internal.command.DbMigrate      : Schema `mysql` is up to date. No migration necessary.
2022-11-03 05:28:42.851  INFO 1 --- [           main] o.hibernate.jpa.internal.util.LogHelper  : HHH000204: Processing PersistenceUnitInfo [name: default]
2022-11-03 05:28:43.152  INFO 1 --- [           main] org.hibernate.Version                    : HHH000412: Hibernate ORM core version 5.4.32.Final
2022-11-03 05:28:44.266  INFO 1 --- [           main] o.hibernate.annotations.common.Version   : HCANN000001: Hibernate Commons Annotations {5.1.2.Final}
2022-11-03 05:28:44.470  INFO 1 --- [           main] org.hibernate.dialect.Dialect            : HHH000400: Using dialect: org.hibernate.dialect.MariaDB53Dialect
2022-11-03 05:28:45.585  INFO 1 --- [           main] o.h.e.t.j.p.i.JtaPlatformInitiator       : HHH000490: Using JtaPlatform implementation: [org.hibernate.engine.transaction.jta.platform.internal.NoJtaPlatform]
2022-11-03 05:28:45.595  INFO 1 --- [           main] j.LocalContainerEntityManagerFactoryBean : Initialized JPA EntityManagerFactory for persistence unit 'default'
2022-11-03 05:28:50.149  INFO 1 --- [           main] o.s.b.c.r.s.JobRepositoryFactoryBean     : No database type set, using meta data indicating: MYSQL
2022-11-03 05:28:50.181  INFO 1 --- [           main] o.s.c.d.s.b.SimpleJobServiceFactoryBean  : No database type set, using meta data indicating: MYSQL
2022-11-03 05:28:50.570  WARN 1 --- [           main] JpaBaseConfiguration$JpaWebConfiguration : spring.jpa.open-in-view is enabled by default. Therefore, database queries may be performed during view rendering. Explicitly configure spring.jpa.open-in-view to disable this warning
2022-11-03 05:28:55.262  INFO 1 --- [           main] .s.c.DataFlowControllerAutoConfiguration : Skipper URI [http://10.233.54.109:80/api]
2022-11-03 05:28:56.061  INFO 1 --- [           main] o.a.coyote.http11.Http11NioProtocol      : Starting ProtocolHandler ["http-nio-80"]
2022-11-03 05:28:56.067  WARN 1 --- [           main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.context.ApplicationContextException: Failed to start bean 'webServerStartStop'; nested exception is org.springframework.boot.web.server.WebServerException: Unable to start embedded Tomcat server
2022-11-03 05:28:56.367  INFO 1 --- [           main] j.LocalContainerEntityManagerFactoryBean : Closing JPA EntityManagerFactory for persistence unit 'default'
2022-11-03 05:28:56.368  INFO 1 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Shutdown initiated...
2022-11-03 05:28:56.457  INFO 1 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Shutdown completed.
2022-11-03 05:28:56.460  INFO 1 --- [           main] o.a.coyote.http11.Http11NioProtocol      : Pausing ProtocolHandler ["http-nio-80"]
2022-11-03 05:28:56.460  INFO 1 --- [           main] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
2022-11-03 05:28:56.464  INFO 1 --- [           main] o.a.coyote.http11.Http11NioProtocol      : Stopping ProtocolHandler ["http-nio-80"]
2022-11-03 05:28:56.464  INFO 1 --- [           main] o.a.coyote.http11.Http11NioProtocol      : Destroying ProtocolHandler ["http-nio-80"]
2022-11-03 05:28:56.958 ERROR 1 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.context.ApplicationContextException: Failed to start bean 'webServerStartStop'; nested exception is org.springframework.boot.web.server.WebServerException: Unable to start embedded Tomcat server
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:181)
	at org.springframework.context.support.DefaultLifecycleProcessor.access$200(DefaultLifecycleProcessor.java:54)
	at org.springframework.context.support.DefaultLifecycleProcessor$LifecycleGroup.start(DefaultLifecycleProcessor.java:356)
	at java.base/java.lang.Iterable.forEach(Unknown Source)
	at org.springframework.context.support.DefaultLifecycleProcessor.startBeans(DefaultLifecycleProcessor.java:155)
	at org.springframework.context.support.DefaultLifecycleProcessor.onRefresh(DefaultLifecycleProcessor.java:123)
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:935)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:586)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:145)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:434)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:338)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1343)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1332)
	at org.springframework.cloud.dataflow.server.single.DataFlowServerApplication.main(DataFlowServerApplication.java:52)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88)
Caused by: org.springframework.boot.web.server.WebServerException: Unable to start embedded Tomcat server
	at org.springframework.boot.web.embedded.tomcat.TomcatWebServer.start(TomcatWebServer.java:229)
	at org.springframework.boot.web.servlet.context.WebServerStartStopLifecycle.start(WebServerStartStopLifecycle.java:43)
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:178)
	... 22 common frames omitted
Caused by: java.lang.IllegalArgumentException: standardService.connector.startFailed
	at org.apache.catalina.core.StandardService.addConnector(StandardService.java:238)
	at org.springframework.boot.web.embedded.tomcat.TomcatWebServer.addPreviouslyRemovedConnectors(TomcatWebServer.java:282)
	at org.springframework.boot.web.embedded.tomcat.TomcatWebServer.start(TomcatWebServer.java:213)
	... 24 common frames omitted
Caused by: org.apache.catalina.LifecycleException: Protocol handler start failed
	at org.apache.catalina.connector.Connector.startInternal(Connector.java:1075)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.StandardService.addConnector(StandardService.java:234)
	... 26 common frames omitted
Caused by: java.net.SocketException: Permission denied
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Unknown Source)
	at java.base/sun.nio.ch.Net.bind(Unknown Source)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(Unknown Source)
	at org.apache.tomcat.util.net.NioEndpoint.initServerSocket(NioEndpoint.java:271)
	at org.apache.tomcat.util.net.NioEndpoint.bind(NioEndpoint.java:226)
	at org.apache.tomcat.util.net.AbstractEndpoint.bindWithCleanup(AbstractEndpoint.java:1208)
	at org.apache.tomcat.util.net.AbstractEndpoint.start(AbstractEndpoint.java:1294)
	at org.apache.coyote.AbstractProtocol.start(AbstractProtocol.java:614)
	at org.apache.catalina.connector.Connector.startInternal(Connector.java:1072)
	... 28 common frames omitted
```

**Solution description:**
Fix containerPort to 8080
```
# /src/kubernetes/server/server-deployment.yaml
...
        - containerPort: 8080
        livenessProbe:
          httpGet:
            path: /management/health
            port: 8080
          initialDelaySeconds: 45
        readinessProbe:
          httpGet:
            path: /management/info
            port: 8080
          initialDelaySeconds: 45
        resources:
          limits:
            cpu: 1.0
            memory: 2048Mi
          requests:
            cpu: 0.5
            memory: 1024Mi
        env:
        - name: KUBERNETES_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: "metadata.namespace"
        - name: SERVER_PORT
          value: '8080'
```
```
# /src/kubernetes/server/server-svc.yml
...
  ports:
    - port: 80
      targetPort: 8080
      name: scdf-server
....
```

**Description of alternatives:**

**Additional context:**
A non-root user cannot run application with 80 port.

--- a/src/carvel/config/dataflow-deployment.yml
+++ b/src/carvel/config/dataflow-deployment.yml
@@ -44,7 +44,7 @@ spec:
           mountPath: /workspace/runtime/secrets
           readOnly: true
         ports:
-        - containerPort: 80
+        - containerPort: 9393
         startupProbe:
           httpGet:
             port: 9393

--- a/src/carvel/config/skipper-deployment.yml
+++ b/src/carvel/config/skipper-deployment.yml
@@ -42,7 +42,7 @@ spec:
           readOnly: true
         #@ end
         ports:
-        - containerPort: 80
+        - containerPort: 7577
         startupProbe:
           httpGet:
             port: 7577

--- a/src/kubernetes/server/server-deployment.yaml
+++ b/src/kubernetes/server/server-deployment.yaml
@@ -26,20 +26,20 @@ spec:
             mountPath: /etc/secrets/database
             readOnly: true
         ports:
-        - containerPort: 80
+        - containerPort: 9393
         livenessProbe:
           httpGet:
             path: /management/health
-            port: 80
+            port: 9393
           initialDelaySeconds: 45
         readinessProbe:
           httpGet:
             path: /management/info
-            port: 80
+            port: 9393
           initialDelaySeconds: 45
         startupProbe:
           tcpSocket:
-            port: 80
+            port: 9393
           initialDelaySeconds: 15
           periodSeconds: 3
           failureThreshold: 120
@@ -60,7 +60,7 @@ spec:
             fieldRef:
               fieldPath: "metadata.namespace"
         - name: SERVER_PORT
-          value: '80'
+          value: '9393'
         - name: SPRING_CLOUD_CONFIG_ENABLED
           value: 'false'
         - name: SPRING_CLOUD_DATAFLOW_FEATURES_ANALYTICS_ENABLED

--- a/src/kubernetes/server/server-svc.yaml
+++ b/src/kubernetes/server/server-svc.yaml
@@ -10,6 +10,7 @@ spec:
   type: LoadBalancer
   ports:
     - port: 80
+      targetPort: 9393
       name: scdf-server
   selector:
     app: scdf-server

--- a/src/kubernetes/skipper/skipper-deployment.yaml
+++ b/src/kubernetes/skipper/skipper-deployment.yaml
@@ -26,7 +26,7 @@ spec:
             mountPath: /etc/secrets/database
             readOnly: true
         ports:
-        - containerPort: 80
+        - containerPort: 7577
         livenessProbe:
           httpGet:
             path: /actuator/health
@@ -59,6 +59,8 @@ spec:
           value: 'en_US.utf8'
         - name: JDK_JAVA_OPTIONS
           value: '-Dfile.encoding=UTF-8 -Dsun.jnu.encoding=UTF-8'
+        - name: SERVER_PORT
+          value: '7577'
         - name: SPRING_CLOUD_CONFIG_ENABLED
           value: 'false'
         - name: SPRING_CLOUD_KUBERNETES_CONFIG_ENABLE_API

--- a/src/templates/kubernetes/server/server-deployment.yaml
+++ b/src/templates/kubernetes/server/server-deployment.yaml
@@ -26,16 +26,16 @@ spec:
             mountPath: /etc/secrets/database
             readOnly: true
         ports:
-        - containerPort: 80
+        - containerPort: 8080
         livenessProbe:
           httpGet:
             path: /management/health
-            port: 80
+            port: 8080
           initialDelaySeconds: 45
         readinessProbe:
           httpGet:
             path: /management/info
-            port: 80
+            port: 8080
           initialDelaySeconds: 45
         resources:
           limits:
@@ -56,7 +56,7 @@ spec:
               fieldRef:
                 fieldPath: "metadata.namespace"
           - name: SERVER_PORT
-            value: '80'
+            value: '8080'
           - name: SPRING_CLOUD_CONFIG_ENABLED
             value: 'false'
           - name: SPRING_CLOUD_DATAFLOW_FEATURES_ANALYTICS_ENABLED

--- a/src/templates/kubernetes/server/server-svc.yaml
+++ b/src/templates/kubernetes/server/server-svc.yaml
@@ -10,6 +10,7 @@ spec:
   type: LoadBalancer
   ports:
     - port: 80
+      targetPort: 8080
       name: scdf-server
   selector:
     app: scdf-server

--- a/src/templates/kubernetes/skipper/skipper-deployment.yaml
+++ b/src/templates/kubernetes/skipper/skipper-deployment.yaml
@@ -26,7 +26,7 @@ spec:
             mountPath: /etc/secrets/database
             readOnly: true
         ports:
-        - containerPort: 80
+        - containerPort: 7577
         livenessProbe:
           httpGet:
             path: /actuator/health

